.github/workflows/deploy.yml

```diff
@@ -0,0 +1,134 @@
+      with:
+        toolchain: stable
+
+    - name: Build cargo-careful
+      run: |
+        cargo build --release
+
+    - name: Get the version
+      shell: bash
+      id: tagName
+      run: |
+        VERSION=$(cargo pkgid | cut -d# -f2)
+        echo "tag=$VERSION" >> $GITHUB_OUTPUT
+
+    - name: Build package
+      id: package
+      shell: bash
+      run: |
+        echo "file=./target/release/cargo-careful.exe" >> $GITHUB_OUTPUT
+        echo "name=cargo-careful.exe" >> $GITHUB_OUTPUT
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ steps.package.outputs.name }}
+        path: ${{ steps.package.outputs.file }}
+
+  create-unix-binaries:
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+          - os: macos-latest
+            target: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install Rust stable
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: stable
+        target: ${{ matrix.target }}
+
+    - name: Install musl
+      if: contains(matrix.target, 'linux-musl')
+      run: |
+        sudo apt-get install musl-tools
+
+    - name: Build cargo-careful
+      run: |
+        # TODO: Remember to add RUSTFLAGS=+crt-static for musl target when
+        # static linkage will not be the default behaviour
+        cargo build --release --target ${{ matrix.target }}
+
+    - name: Strip binary
+      run: |
+        strip target/${{ matrix.target }}/release/cargo-careful
+
+    - name: Get the version
+      id: tagName
+      run: |
+        VERSION=$(cargo pkgid | cut -d# -f2)
+        echo "tag=$VERSION" >> $GITHUB_OUTPUT
+
+    - name: Build package
+      id: package
+      run: |
+        echo "file=./target/${{ matrix.target }}/release/cargo-careful" >> $GITHUB_OUTPUT
+        echo "name=cargo-careful-${{ matrix.target }}" >> $GITHUB_OUTPUT
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ steps.package.outputs.name }}
+        path: ${{ steps.package.outputs.file }}
+
+
+  deploy:
+
+    needs: [create-windows-binaries, create-unix-binaries]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Get version
+        id: tagName
+        run: |
+          VERSION=$(cargo pkgid | cut -d# -f2)
+          echo "tag=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: ./binaries
+
+      - name: Create a release
+        uses: softprops/action-gh-release@v1
+        with:
+          name: v${{ steps.tagName.outputs.tag }}
```

Comment:
It looks like this defaults to the name of the tag... so we could probably avoid those `tagName` jobs by just removing this entirely? The tag name should always match the cargo package version for a release.

If anything we might want to put a little check somewhere which ensures that the tag name and the cargo version match.

Reply:
This parameter assigns a name to a release which corresponds to `v{tag}`, Do you want this part completely removed or use the cargo package version?

Reply:
What I am saying is that, reading the docs for `softprops/action-gh-release`, the `name` field defaults to the current tag. So what is the point of assigning it explicitly here?

Reply:
Oh, you're perfectly right.
- Removed that field
- Add a check to compare crate tag and `Github` tag